src/clients/BundleDataClient/BundleDataClient.ts

```diff
@@ -804,24 +809,24 @@ export class BundleDataClient {
         if (originChainId === destinationChainId) {
           continue;
         }
-        originClient.getDepositsForDestinationChain(destinationChainId).forEach((deposit) => {
-          if (isZeroValueDeposit(deposit)) {
+        originClient.getDepositsForDestinationChainWithDuplicates(destinationChainId).forEach((deposit) => {
+          // Only evaluate deposits that are in this bundle or in previous bundles. This means we cannot issue fill
+          // refunds or slow fills here for deposits that are in future bundles (i.e. "pre-fills"). Instead, we'll
+          // evaluate these pre-fills once the deposit is inside the "current" bundle block range.
+          if (deposit.blockNumber > originChainBlockRange[1] || isZeroValueDeposit(deposit)) {
             return;
           }
           depositCounter++;
           const relayDataHash = this.getRelayHashFromEvent(deposit);
-          if (v3RelayHashes[relayDataHash]) {
-            // If we've seen this deposit before, then skip this deposit. This can happen if our RPC provider
-            // gives us bad data.
-            return;
+
+          // Duplicate deposits are treated like normal deposits.
+          if (!v3RelayHashes[relayDataHash]) {
```

Comment:
we can probably just remove this if statement since its ok to overwrite the deposit. Currently it'll always get set to the oldest (the first one returned by getDepositsForDestinationChainWithDuplicates() and I also think that's ok